File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        # Count occurrences of each string
        count = Counter(arr)
        
        # Collect distinct strings
        distinct_strings = [string for string in arr if count[string] == 1]
        
        # Return the kth distinct string, if it exists
        if k <= len(distinct_strings):
            return distinct_strings[k - 1]  # k is 1-indexed
        else:
            return ""  # Not enough distinct strings

File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.js
/**
 * @param {string[]} arr
 * @param {number} k
 * @return {string}
 */
var kthDistinct = function(arr, k) {
    const count = {}; // Object to count occurrences of each string
    
    // Count occurrences of each string
    for (const string of arr) {
        count[string] = (count[string] || 0) + 1;
    }

    // Collect distinct strings
    const distinctStrings = [];
    for (const string of arr) {
        if (count[string] === 1) { // Only consider strings that appear once
            distinctStrings.push(string);
        }
    }

    // Return the kth distinct string, if it exists
    if (k <= distinctStrings.length) {
        return distinctStrings[k - 1]; // k is 1-indexed
    } else {
        return ""; // Not enough distinct strings
    }
};
